opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/AggregationQueryBuilder.java

```diff
@@ -104,14 +105,14 @@ public AggregationQueryBuilder(
   }
 
   /**
-   * Build ElasticsearchExprValueFactory.
+   * Build mapping for OpenSearchExprValueFactory.
    */
-  public Map<String, ExprType> buildTypeMapping(
+  public Map<String, MappingEntry> buildTypeMapping(
       List<NamedAggregator> namedAggregatorList,
       List<NamedExpression> groupByList) {
-    ImmutableMap.Builder<String, ExprType> builder = new ImmutableMap.Builder<>();
-    namedAggregatorList.forEach(agg -> builder.put(agg.getName(), agg.type()));
-    groupByList.forEach(group -> builder.put(group.getNameOrAlias(), group.type()));
+    ImmutableMap.Builder<String, MappingEntry> builder = new ImmutableMap.Builder<>();
+    namedAggregatorList.forEach(agg -> builder.put(agg.getName(), new MappingEntry(null, null, agg.type())));
+    groupByList.forEach(group -> builder.put(group.getNameOrAlias(), new MappingEntry(null, null, group.type())));
```

Comment:
I'm thinking should we refactor `OpenSearchDataType`? Rather than static `Enum`, we instantiate it with mapping info for each field during analyzing.

Reply:
@dai-chen,
Please see changes in #180. They don't contain date format processing, but only refactor of `OpenSearchDataType`.